src/Akka.Streams.Kafka.Tests/BugFix240SupervisionStrategy.cs

```diff
@@ -0,0 +1,434 @@
+                .RunWith(KafkaProducer.PlainSink(ProducerSettings), Materializer);
+
+            var drainingControl = KafkaConsumer.CommittableSource(consumerSettings, Subscriptions.Assignment(topicPartition))
+                .Via(Flow.Create<CommittableMessage<Null, string>>().Select(x =>
+                {
+                    counter++;
+                    if (counter % 5 == 0)
+                        throw new Exception("BOOM!");
+                    return x;
+                }))
+                .WithAttributes(Attributes.CreateName("CommitableSource").And(ActorAttributes.CreateSupervisionStrategy(Decider)))
+                .Select(c => (c.Record.Topic, c.Record.Message.Value, c.CommitableOffset))
+                .SelectAsync(1, async t =>
+                {
+                    Log.Info($"[{t.Topic}]: {t.Value}");
+                    // simulate a request-response call that takes 10ms to complete here
+                    await Task.Delay(10);
+                    return t;
+                })
+                .Select(t => ProducerMessage.Single(new ProducerRecord<Null, string>($"{t.Topic}-done", t.Value),
+                    t.CommitableOffset))
+                .Via(KafkaProducer.FlexiFlow<Null, string, ICommittableOffset>(ProducerSettings)).WithAttributes(Attributes.CreateName("FlexiFlow"))
+                .Select(m => (ICommittable)m.PassThrough)
+                .AlsoToMaterialized(Committer.Sink(committerSettings), DrainingControl<NotUsed>.Create)
+                .To(Flow.Create<ICommittable>()
+                    .Async()
+                    .GroupedWithin(1000, TimeSpan.FromSeconds(1))
+                    .Select(c => c.Count())
+                    .Log("MsgCount").AddAttributes(Attributes.CreateLogLevels(LogLevel.InfoLevel))
+                    .To(Sink.Ignore<int>()))
+                .Run(Sys.Materializer());
+
+            await Task.Delay(TimeSpan.FromSeconds(2));
+            await drainingControl.Shutdown();
+            callCount.Should().BeGreaterThan(0);
+        }
+        
+        [Fact]
+        public async Task SupervisionStrategy_Decider_on_PlainSink_should_work()
+        {
+            var callCount = 0;
+            Directive Decider(Exception cause)
+            {
+                callCount++;
+                switch (cause)
+                {
+                    case ProduceException<Null, string> ex when ex.Error.IsSerializationError():
+                        return Directive.Resume;
+                    default:
+                        return Directive.Stop;
+                }
+            }
+
+            var topic1 = CreateTopic(1);
+            var group1 = CreateGroup(1);
+            
+            var producerSettings = ProducerSettings<Null, string>
+                .Create(Sys, null, new FailingSerializer())
+                .WithBootstrapServers(Fixture.KafkaServer);
+            
+            var sourceTask = Source
```

Comment:
Add comments here to explain how failures are injected

Reply:
Done